src/derivative_wrappers.jl

```diff
@@ -40,55 +40,104 @@ function jacobian(f, x, integrator)
+    tmp = N + 1
+  else
+    tmp = 2N
+  end
+  tmp
+end
 
 jacobian_finitediff(f, x, diff_type, integrator) =
-    DiffEqDiffTools.finite_difference_derivative(f, x, diff_type, eltype(x), dir = diffdir(integrator))
+    (DiffEqDiffTools.finite_difference_derivative(f, x, diff_type, eltype(x), dir = diffdir(integrator)),2)
 jacobian_finitediff(f, x::AbstractArray, diff_type, integrator) =
-    DiffEqDiffTools.finite_difference_jacobian(f, x, diff_type, eltype(x), Val{false}, dir = diffdir(integrator))
+    (DiffEqDiffTools.finite_difference_jacobian(f, x, diff_type, eltype(x), Val{false}, 
+      dir = diffdir(integrator)),_nfcount(length(x),diff_type))
+jacobian_finitediff(f, x::AbstractArray, diff_type, integrator, colorvec) =
+    (DiffEqDiffTools.finite_difference_jacobian(f, x, diff_type, eltype(x), Val{false}, 
+      dir = diffdir(integrator), color = colorvec, sparsity = integrator.f.jac_prototype),_nfcount(maximum(colorvec),diff_type))
+
+jacobian_finitediff_forward!(J,f,x,jac_config,forwardcache,integrator)=(DiffEqDiffTools.finite_difference_jacobian!(J,f,x,jac_config,forwardcache,dir=diffdir(integrator));length(x))
+jacobian_finitediff_forward!(J,f,x,jac_config,forwardcache,integrator,colorvec)=
+  (DiffEqDiffTools.finite_difference_jacobian!(J,f,x,jac_config,forwardcache,
+    dir=diffdir(integrator),color=colorvec,sparsity=integrator.f.jac_prototype);maximum(colorvec))
+jacobian_finitediff!(J,f,x,jac_config,integrator)=(DiffEqDiffTools.finite_difference_jacobian!(J,f,x,jac_config,dir=diffdir(integrator));2*length(x))
+jacobian_finitediff!(J,f,x,jac_config,integrator,colorvec)=
+  (DiffEqDiffTools.finite_difference_jacobian!(J,f,x,jac_config,
+    dir=diffdir(integrator),color=colorvec,sparsity=integrator.f.jac_prototype);2*maximum(colorvec))
+jacobian_autodiff!(J,f,x,jac_config)=forwarddiff_color_jacobian!(J,f,x,jac_config)#J::SparseMatrixCSC
+jacobian_autodiff!(J,f,fx,x,jac_config)=ForwardDiff.jacobian!(J,f,fx,x,jac_config)
 
 function jacobian!(J::AbstractMatrix{<:Number}, f, x::AbstractArray{<:Number}, fx::AbstractArray{<:Number}, integrator::DiffEqBase.DEIntegrator, jac_config)
     alg = unwrap_alg(integrator, true)
     if alg_autodiff(alg)
-      ForwardDiff.jacobian!(J, f, fx, x, jac_config)
+      if DiffEqBase.has_colorvec(integrator.f)
+        jacobian_autodiff!(J,f,x,jac_config)
+      else
+        jacobian_autodiff!(J,f,fx,x,jac_config)
+      end
       integrator.destats.nf += 1
     else
       isforward = alg.diff_type === Val{:forward}
       if isforward
         forwardcache = get_tmp_cache(integrator, alg, unwrap_cache(integrator, true))[2]
         f(forwardcache, x)
         integrator.destats.nf += 1
-        DiffEqDiffTools.finite_difference_jacobian!(J, f, x, jac_config, forwardcache, dir = diffdir(integrator))
+        if DiffEqBase.has_colorvec(integrator.f)
+          tmp=jacobian_finitediff_forward!(J, f, x, jac_config, forwardcache, integrator, integrator.f.colorvec)
```

Comment:
can get rid of the branch and just dispatch on `::Nothing`

Reply:
Which branch?